shimmy/dm_lab_compatibility.py

```diff
@@ -0,0 +1,97 @@
+"""Wrapper to convert a dm_lab environment into a gymnasium compatible environment.
+
+Taken from
+https://github.com/ikostrikov/dmcgym/blob/main/dmcgym/env.py
+and modified to modern gymnasium API
+"""
+from __future__ import annotations
+
+from typing import Any
+
+import gymnasium
+import numpy as np
+from dm_env import Environment
+from gymnasium.core import ObsType
+
+from shimmy.utils.dm_lab_utils import dm_lab_spec2gym_space
+
+
+class DmLabCompatibility(gymnasium.Env[ObsType, np.ndarray]):
+    """A compatibility wrapper that converts a dm_lab-control environment into a gymnasium environment."""
+
+    metadata = {"render_modes": ["human", "rgb_array"], "render_fps": 10}
+
+    def __init__(
+        self,
+        env: Environment,
+        render_mode: str | None = None,
+        render_height: int = 84,
+        render_width: int = 84,
+        camera_id: int = 0,
+    ):
+        """Initialises the environment with a render mode along with render information."""
+        self._env = env
+
+        self.observation_space = dm_lab_spec2gym_space(env.observation_spec())
+        self.action_space = dm_lab_spec2gym_space(env.action_spec())
+
+        assert render_mode is None or render_mode in self.metadata["render_modes"]
+        self.render_mode = render_mode
+        self.render_height, self.render_width = render_height, render_width
+        self.camera_id = camera_id
+
+        if self.render_mode == "human":
+            from gymnasium.envs.mujoco.mujoco_rendering import Viewer
+
+            self.viewer = Viewer(
+                self._env.physics.model.ptr, self._env.physics.data.ptr
+            )
+
+    def reset(
+        self, *, seed: int | None = None, options: dict[str, Any] | None = None
+    ) -> tuple[ObsType, dict[str, Any]]:
+        """Resets the dm-lab environment."""
+        super().reset(seed=seed)
+
+        timestep = self._env.reset()
+
+        obs, reward, terminated, truncated, info = expose_timestep(timestep)
+
+        obs = None
```

Comment:
??

Reply:
Working on it, this is not finished code